kit/Kit.cpp

```diff
@@ -1152,6 +1152,17 @@ class Document final : public DocumentManagerInterface
             }
             return;
         }
+        else if (type == LOK_CALLBACK_DOCUMENT_PASSWORD_RESET) {
+            Document* document = dynamic_cast<Document*>(descriptor->getDoc());
+            if (document)
+            {
+                document->_isDocPasswordProtected = p ? !payload.empty() : false;
```

Comment:
Isn't it simpler like this?
`document->_isDocPasswordProtected = p && !payload.empty();`